iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -646,13 +662,32 @@ public override async Task<long> SendTwinPatchAsync(TwinCollection reportedPrope
             {
                 // No matter what, remove the requestId from this dictionary since no thread will be waiting for it anymore
                 _reportedPropertyUpdateResponseCompletions.TryRemove(requestId, out TaskCompletionSource<PatchTwinResponse> _);
+                _twinResponseTimeouts.TryRemove(requestId, out DateTimeOffset _);
+            }
+        }
+
+        private void CheckTimeout(object _)
+        {
+            if (_twinResponseTimeouts.Any())
+            {
+                var currentDateTime = DateTime.UtcNow;
+                foreach (KeyValuePair<string, DateTimeOffset> entry in _twinResponseTimeouts)
+                {
+                    if (currentDateTime - entry.Value > s_twinResponseTimeout)
+                    {
+                        _getTwinResponseCompletions.TryRemove(entry.Key, out TaskCompletionSource<GetTwinResponse> _);
+                        _reportedPropertyUpdateResponseCompletions.TryRemove(entry.Key, out TaskCompletionSource<PatchTwinResponse> _);
+                        _twinResponseTimeouts.TryRemove(entry.Key, out DateTimeOffset _);
+                    }
+                }
             }
```

Comment:
If you handle error cases early, then the rest of the code can be outdented (line length doesn't get impacted as much.
```suggestion
            if (!_twinResponseTimeouts.Any())
            {
                return;
            }
            
            var currentDateTime = DateTime.UtcNow;
            foreach (KeyValuePair<string, DateTimeOffset> entry in _twinResponseTimeouts)
            {
                if (currentDateTime - entry.Value > s_twinResponseTimeout)
                {
                    _getTwinResponseCompletions.TryRemove(entry.Key, out TaskCompletionSource<GetTwinResponse> _);
                    _reportedPropertyUpdateResponseCompletions.TryRemove(entry.Key, out TaskCompletionSource<PatchTwinResponse> _);
                    _twinResponseTimeouts.TryRemove(entry.Key, out DateTimeOffset _);
                }
            }
```